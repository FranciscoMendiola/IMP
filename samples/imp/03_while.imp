while x <= 100 do
  x := x + 1
