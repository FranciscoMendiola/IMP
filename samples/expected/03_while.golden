Token {tokenKind = KWhile, lexeme = "while", position = 0}
Token {tokenKind = Id, lexeme = "x", position = 6}
Token {tokenKind = Leq, lexeme = "<=", position = 8}
Token {tokenKind = IntLit, lexeme = "100", position = 11}
Token {tokenKind = KDo, lexeme = "do", position = 15}
Token {tokenKind = Id, lexeme = "x", position = 20}
Token {tokenKind = Assign, lexeme = ":=", position = 22}
Token {tokenKind = Id, lexeme = "x", position = 25}
Token {tokenKind = Plus, lexeme = "+", position = 27}
Token {tokenKind = IntLit, lexeme = "1", position = 29}
